files/private-chef-ctl-commands/helpers/key_ctl_helper.rb

```diff
@@ -0,0 +1,165 @@
+  #
+  # For example, if you have two arguments with mandatory input, --username NAME and
+  # --public-key-path PATH, defined in optparse, and you pass:
+  #
+  # cmd --username --public-key-path /etc/key
+  #
+  # It will detect the input to --username as --public-key-path and won't even detect --public-key-path as an argument.
+  # This helper method takes in a list of argument commands and checks that the input to a given argument isn't actually
+  # another argument, and throws an error with a relevant message if so, fixing the case that an argument with mandatory
+  # input isn't actually eating up the next argument as its input.
+  def catch_argument_passed_as_input(arg_list, arg, arg_input)
+    arg_list.each do |arg_element|
+      if arg_input.strip == arg_element
+        exit_failure(missing_input_msg(arg))
+      end
+    end
+  end
+
+  def exit_failure(msg)
+    raise SystemExit.new(1, msg)
+  end
+
+  def get_rest(url)
+    @chef_rest.get_rest(url)
+  end
+
+  def delete_rest(url)
+    @chef_rest.delete_rest(url)
+  end
+
+  def post_rest(url, body)
+    @chef_rest.post_rest
+  end
+
+  def pivotal_config
+    "/etc/opscode/pivotal.rb"
+  end
+
+  def configure_chef_rest
+    Chef::Config.from_file(pivotal_config)
+    Chef::REST.new(Chef::Config[:chef_server_root])
+  end
+
+  def check_valid_iso_date(expiration_date)
+    unless /^(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z|infinity)$/.match(expiration_date)
+      exit_failure(invalid_date_msg)
+    end
+  end
+
+  def read_and_check_key(key_path)
+    begin
+      key = File.read(key_path)
+    rescue
+      exit_failure(public_key_path_msg)
+    end
+
+    unless /BEGIN (RSA |)PUBLIC KEY/.match(key)
+      exit_failure(not_a_public_key_msg)
+    end
+    key
+  end
+
+  def generate_fingerprint(key)
+    begin
+      openssl_key_object = OpenSSL::PKey::RSA.new(key)
+      data_string = OpenSSL::ASN1::Sequence([
+                                              OpenSSL::ASN1::Integer.new(openssl_key_object.public_key.n),
+                                              OpenSSL::ASN1::Integer.new(openssl_key_object.public_key.e)
+                                            ])
+      OpenSSL::Digest::SHA1.hexdigest(data_string.to_der).scan(/../).join(':')
+    rescue
+      exit_failure(cannot_generate_fingerprint_msg)
+    end
+  end
+
+  def add_key_usage
+    "\nUsage: Expiration date defaults to infinity. Pass an ISO 8601 fomatted string: YYYY-MM-DDTHH:MM:SSZ e.g. 2013-12-24T21:00:00Z in UTC timezone\nUsage: Default name used is the fingerprint of the key passed."
+  end
+
+  def parse_missing_arg_error(err)
+    arg = err.message.match(/(?<=missing argument:\s).*/)[0]
+    exit_failure(missing_valid_input_msg(arg))
+  end
+
+  def parse_invalid_arg_error(err)
+    arg = err.message.match(/(?<=invalid option:\s).*/)[0]
+    exit_failure(invalid_arg_msg(arg))
+  end
+
+  def exit_http_fail(err)
```

Comment:
I just meant capturing and displaying the error message itself, instead of dumping the json.